src/Sarif/Core/Run.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public partial class Run
+    {
+        public bool ShouldSerializeInvocations() { return this.Invocations != null && this.Invocations.Where((s) => s != null).FirstOrDefault() != null; }
```

Comment:
>this.Invocations.Where((s) => s != null).FirstOrDefault() != null [](start = 86, length = 65)

Again, use `Any` with predicate. Won't continue to comment on this issue.

Reply:
You can also do this in places where the code checks for `Count() > 0`.

---
In reply to: [229091708](https://github.com/Microsoft/sarif-sdk/pull/1121#discussion_r229091708) [](ancestors = 229091708)